artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPSessionContext.java

```diff
@@ -229,8 +226,7 @@ public void addReplicaTarget(Receiver receiver) throws Exception {
          AMQPMirrorControllerTarget protonReceiver = new AMQPMirrorControllerTarget(sessionSPI, connection, this, receiver, server);
          protonReceiver.initialize();
          receivers.put(receiver, protonReceiver);
-         ServerProducer serverProducer = new ServerProducerImpl(receiver.getName(), "AMQP", receiver.getTarget().getAddress());
-         sessionSPI.addProducer(serverProducer);
+         sessionSPI.addProducer(receiver.getName(), receiver.getTarget().getAddress());
```

Comment:
This addProducer call is fed getName(), while the addProducer call changed below uses getUniqueName(), seems inconsistent.

Reply:
fixed